libs/ledger/src/executor.cpp

```diff
@@ -365,11 +359,7 @@ bool Executor::ExecuteTransactionContract(Result &result)
                       scaled_charge, " (base: ", base_charge, " storage: ", storage_charge,
                       " compute: ", compute_charge, " shards: ", allowed_shards_.PopCount(), ")");
 
-      // create wealth transactions are always free
-      if (!IsCreateWealth(*current_tx_))
-      {
-        result.charge += scaled_charge;
-      }
+      result.charge += scaled_charge;
```

Comment:
Shouldn't this be capped at the charge limit?

Reply:
I think that in this instance the result.charge is what it has cost, so it can and will exceed the charge limit (the amount the TX is willing to pay), in which case your TX fails. Second opinion @ejfitzgerald, since I don't really want to get this wrong.

Reply:
My thinking was that the information of insufficient charge is already conveyed by `result.status = Status::INSUFFICIENT_CHARGE;` below. Obviously you'd need to rearrange this function a bit, but overall setting `result.charge` above the charge limit seems runs a needless risk of overcharging someone accidentally.